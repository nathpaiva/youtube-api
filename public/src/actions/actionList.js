'user strict';

export function listVideos(list) {
  return { type: 'LIST', list };
}

export function listOneVideos(list) {
  return { type: 'SELECTFIRST', list };
}

export function selectOneVideo(list) {
  return { type: 'SELECTONE', list };
}
