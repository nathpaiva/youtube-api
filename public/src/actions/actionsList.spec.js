'use stict';

import React from 'react';
import renderer from 'react-test-renderer';
import { listVideos, listOneVideos, selectOneVideo } from './actionList';


describe('#Test actions', () => {
  it('#Should check list videos', () => {
    const response = listVideos([]);
    expect(response.type === 'LIST').toEqual(true);

    expect(response).toMatchSnapshot();
  });

  it('#Should check list one videos', () => {
    const response = listOneVideos([]);
    expect(response.type === 'SELECTFIRST').toEqual(true);

    expect(response).toMatchSnapshot();
  });

  it('#Should check list one videos', () => {
    const response = selectOneVideo([]);
    expect(response.type === 'SELECTONE').toEqual(true);

    expect(response).toMatchSnapshot();
  });
});
