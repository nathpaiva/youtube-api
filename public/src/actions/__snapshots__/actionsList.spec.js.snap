// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test actions #Should check list one videos 1`] = `
Object {
  "list": Array [],
  "type": "SELECTFIRST",
}
`;

exports[`#Test actions #Should check list one videos 2`] = `
Object {
  "list": Array [],
  "type": "SELECTONE",
}
`;

exports[`#Test actions #Should check list videos 1`] = `
Object {
  "list": Array [],
  "type": "LIST",
}
`;
